src/main/java/fmartin1/controller/PokedexController.java

```diff
@@ -0,0 +1,79 @@
+package fmartin1.controller;
+
+
+import fmartin1.model.pokemon.Pokemon;
+import fmartin1.model.pokemon.PokemonComparator;
+import fmartin1.service.PokedexService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.stream.Collectors;
+
+@SuppressWarnings("unused")
+@Controller
+@RequestMapping("pokemon")
+public class PokedexController {
+    private final PokedexService _pokedexService;
+
+    @Autowired
+    public PokedexController(PokedexService pokedexService) {
+        _pokedexService = pokedexService;
+    }
+
+    @RequestMapping(value = "/", method = RequestMethod.GET, produces = MediaType.TEXT_PLAIN_VALUE)
+    @ResponseBody
+    public String getAllPokemon(@RequestParam(value = "limit", required = false) Integer limit) {
```

Comment:
I'm thinking about split this method in 2; the first method receives a parameter `limit` and the second one uses the first method but passes the total number of Pokemon

`getPaginatedPokemonList(Integer limit)`
`getAllPokemonList()`
`private static final MAX_POKEMON_LIST = 802`

What do you think?